packages/@tinacms/forms/src/form.test.ts

```diff
@@ -36,6 +37,67 @@ describe('Form', () => {
         new Form(DEFAULTS)
       })
     })
+    describe('without loadInitialValues', () => {
+      it('is not loading', () => {
+        const form = new Form(DEFAULTS)
+
+        expect(form.loading).toBeFalsy()
+      })
+    })
+    describe('with loadInitialValues', () => {
+      describe('before promise resolves', () => {
+        it('is is loading', () => {
+          const form = new Form({
+            ...DEFAULTS,
+            loadInitialValues: () => new Promise(() => {}),
+          })
+
+          expect(form.loading).toBeTruthy()
+        })
+        it('has values === undefined', () => {
+          const form = new Form({
+            ...DEFAULTS,
+            loadInitialValues: () => new Promise(() => {}),
+          })
+
+          expect(form.values).toBeUndefined()
+        })
+      })
+      describe('after promise resolves', () => {
+        it('is not loading once resolved', async () => {
+          let resolve: () => void | null
+
+          const form = new Form({
+            ...DEFAULTS,
+            loadInitialValues: () =>
+              new Promise(r => {
+                resolve = r
```

Comment:
:exploding_head:

Reply:
Shocked by the crazy, resolve a promise outside of the promise shenanigans?

Reply:
I just never would have thought of it. clever